cmd/kube-controller-manager/app/core.go

```diff
@@ -71,9 +71,23 @@ func startServiceController(ctx ControllerContext) (bool, error) {
 }
 
 func startNodeController(ctx ControllerContext) (bool, error) {
-	_, clusterCIDR, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
-	if err != nil {
-		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+	var clusterCIDRs []*net.IPNet
+	for _, cidr := range ctx.Options.ClusterCIDRs {
+		_, cc, err := net.ParseCIDR(cidr)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR[0], err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+	}
+	NodeCIDRMaskSizes := ctx.Options.NodeCIDRMaskSizes
+	// If the new style was not found, check the old.
+	if len(clusterCIDRs) == 0 {
+		_, cc, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+		NodeCIDRMaskSizes = append(NodeCIDRMaskSizes, int(ctx.Options.NodeCIDRMaskSize))
```

Comment:
so the case would here would be using the old style cidr, but it's possible the user added both the new style mask and old style mask? this would get by this section, but later fail as the number of cidrs does not match the number of masks. I could rearrange it to put the new style check in the else clause of the if len(clusterCIDRs) == 0 { this would prevent the extra mask elements.
On the question of mixing old and new, are you saying we should handle that case as valid? or check for this specific case and reject it?